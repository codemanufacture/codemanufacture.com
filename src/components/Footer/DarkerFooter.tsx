import * as React from 'react'
import styled from 'styled-components'
import { colors, displayDimensions } from '../../theme'

interface DarkerFooterProps {
  children: object
}

const StyledDarkerFooter = styled.div`
  display: flex;
  width: 100%;
  background: ${colors.backgroundDarkerGray};
  margin-top: 30px;
  padding: 30px 0;

  .container {
    display: flex;
    width: 66%;
    margin: auto;

    @media (max-width: ${displayDimensions.tabletSize}) {
      width: 100%;
      flex-direction: column;
      padding: 0 30px;
    }
  }
`

const DarkerFooter: React.FunctionComponent<DarkerFooterProps> = ({
  children,
}) => (
  <StyledDarkerFooter>
    <div className="container">{children}</div>
  </StyledDarkerFooter>
)

export default DarkerFooter
