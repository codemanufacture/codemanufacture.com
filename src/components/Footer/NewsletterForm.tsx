import * as React from 'react'
import styled from 'styled-components'
import { colors, typography, transitions, displayDimensions } from '../../theme'
import addToMailchimp from 'gatsby-plugin-mailchimp'
import Input from '../Input'

interface StyledProps {
  isMailchimpMessageVisible: boolean
}

const StyledFormWrapper = styled<StyledProps, 'div'>('div')`
  min-height: 40px;
  width: 50%;
  margin-left: auto;
  margin-right: 0;
  position: relative;

  @media (max-width: ${displayDimensions.tabletSize}) {
    width: 100%;
  }

  .message-wrapper {
    opacity: ${props => (props.isMailchimpMessageVisible ? 1 : 0)};
    width: calc(100% - 20px);
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
    color: #fff;
    transition: opacity ${transitions.basicTransition};
    font-size: ${typography.paragraphSize};
    pointer-events: none;
  }

  form {
    opacity: ${props => (props.isMailchimpMessageVisible ? 0 : 1)};
    display: flex;
    padding: 0 15px;
    align-items: flex-start;
    pointer-events: ${props =>
      props.isMailchimpMessageVisible ? 'none' : 'initial'};

    input {
      + input {
        margin-left: 15px;
      }
    }

    @media (max-width: ${displayDimensions.tabletSize}) {
      flex-direction: column;
      align-items: initial;

      input {
        + input {
          margin-left: initial;
          margin-top: 10px;
        }
      }
    }
  }

  .btn-submit {
    margin-right: 0;
    padding: 0.375rem 0.75rem;
    border: 1px solid transparent;
    border-radius: 0.25rem;
    background: ${colors.backgroundSuccess};
    color: ${colors.buttonTextColor};
    font-size: ${typography.ctaSize};
    line-height: 1.5;
    transition: background-color ${transitions.basicTransition};
    cursor: pointer;
    outline: none;

    &:hover {
      background: ${colors.backgroundSuccessHover};
    }
  }
`

interface NewsletterFormState {
  isMailchimpMessageVisible: boolean
  mailchimpMessage: string
  userEmail: string
  userName: string
}

interface MailchimpCallData {
  msg: string
  result: string
}

class NewsletterForm extends React.PureComponent<{}, NewsletterFormState> {
  constructor(props: object) {
    super(props)

    this.state = {
      isMailchimpMessageVisible: false,
      mailchimpMessage: ``,
      userEmail: ``,
      userName: ``,
    }
  }

  handleEmailInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ userEmail: e.target.value })
  }

  handleNameInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ userName: e.target.value })
  }

  handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault()
    addToMailchimp(this.state.userEmail, { ['FNAME']: this.state.userName })
      .then((data: MailchimpCallData) => {
        console.log(data)
        this.setState({
          isMailchimpMessageVisible: true,
          mailchimpMessage: data.msg,
        })
      })
      .catch((error: object) => {
        console.log(error)
      })
  }

  render() {
    const {
      isMailchimpMessageVisible,
      mailchimpMessage,
      userEmail,
      userName,
    } = this.state

    return (
      <StyledFormWrapper isMailchimpMessageVisible={isMailchimpMessageVisible}>
        <form onSubmit={e => this.handleSubmit(e)}>
          <Input
            onChange={(e: React.SyntheticEvent) =>
              this.handleNameInputChange(e)
            }
            type="text"
            value={userName}
            name="FNAME"
            className="required"
            id="mce-FNAME"
            placeholder="Enter your name"
          />
          <Input
            onChange={(e: React.SyntheticEvent) =>
              this.handleEmailInputChange(e)
            }
            type="email"
            value={userEmail}
            name="EMAIL"
            className="required email"
            id="mce-EMAIL"
            placeholder="Enter your email"
          />
          <input type="submit" value="Subscribe" className="btn-submit" />
        </form>
        <p className="message-wrapper">{mailchimpMessage}</p>
      </StyledFormWrapper>
    )
  }
}

export default NewsletterForm
