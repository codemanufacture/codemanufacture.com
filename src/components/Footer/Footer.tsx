import * as React from 'react'
import styled from 'styled-components'
import { colors } from '../../theme'
import NewsletterForm from './NewsletterForm'

const StyledFooterWrapper = styled.div`
  display: flex;
  width: 100%;
  background: ${colors.backgroundGray};
  padding: 30px 0;
`

const Footer = () => (
  <StyledFooterWrapper>
    <NewsletterForm />
  </StyledFooterWrapper>
)

export default Footer
