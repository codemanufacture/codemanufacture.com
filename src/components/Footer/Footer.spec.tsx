import * as React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Footer from './Footer'

describe('Footer component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
