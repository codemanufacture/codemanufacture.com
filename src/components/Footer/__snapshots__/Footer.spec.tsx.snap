// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  aria-labelledby="footer-heading"
  tw="bg-neutral-900"
>
  <div
    tw="mx-auto max-w-7xl overflow-hidden py-20 px-6 sm:py-24 lg:px-8"
  >
    <nav
      aria-label="Footer"
      tw="-mb-6 columns-2 sm:flex sm:justify-center sm:space-x-12"
    >
      <div
        tw="pb-6"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={{}}
          tw="text-sm leading-6 text-gray-600 hover:text-gray-200"
        >
          Home
        </a>
      </div>
      <div
        tw="pb-6"
      >
        <a
          href="/#service"
          onClick={[Function]}
          onMouseEnter={[Function]}
          tw="text-sm leading-6 text-gray-600 hover:text-gray-200"
        >
          Service
        </a>
      </div>
      <div
        tw="pb-6"
      >
        <a
          href="/#blog"
          onClick={[Function]}
          onMouseEnter={[Function]}
          tw="text-sm leading-6 text-gray-600 hover:text-gray-200"
        >
          Blog
        </a>
      </div>
      <div
        tw="pb-6"
      >
        <a
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          tw="text-sm leading-6 text-gray-600 hover:text-gray-200"
        >
          Contact
        </a>
      </div>
      <div
        tw="pb-6"
      >
        <a
          href="/privacy-policy"
          onClick={[Function]}
          onMouseEnter={[Function]}
          tw="text-sm leading-6 text-gray-600 hover:text-gray-200"
        >
          Privacy
        </a>
      </div>
    </nav>
    <div
      tw="mt-10 flex justify-center space-x-10"
    >
      <a
        href="https://www.linkedin.com/company/codemanufacture"
        tw="text-gray-400 hover:text-gray-500"
      >
        <span
          tw="sr-only"
        >
          LinkedIn
        </span>
        <div
          tw="h-6 w-6"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.335 18.339H15.67v-4.177c0-.996-.02-2.278-1.39-2.278-1.389 0-1.601 1.084-1.601 2.205v4.25h-2.666V9.75h2.56v1.17h.035c.358-.674 1.228-1.387 2.528-1.387 2.7 0 3.2 1.778 3.2 4.091v4.715zM7.003 8.575a1.546 1.546 0 01-1.548-1.549 1.548 1.548 0 111.547 1.549zm1.336 9.764H5.666V9.75H8.34v8.589zM19.67 3H4.329C3.593 3 3 3.58 3 4.297v15.406C3 20.42 3.594 21 4.328 21h15.338C20.4 21 21 20.42 21 19.703V4.297C21 3.58 20.4 3 19.666 3h.003z"
            />
          </svg>
        </div>
      </a>
      <a
        href="https://twitter.com/codemanufacture"
        tw="text-gray-400 hover:text-gray-500"
      >
        <span
          tw="sr-only"
        >
          Twitter
        </span>
        <div
          tw="h-6 w-6"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.29 20.251c7.547 0 11.675-6.253 11.675-11.675 0-.178 0-.355-.012-.53A8.348 8.348 0 0022 5.92a8.19 8.19 0 01-2.357.646 4.118 4.118 0 001.804-2.27 8.224 8.224 0 01-2.605.996 4.107 4.107 0 00-6.993 3.743 11.65 11.65 0 01-8.457-4.287 4.106 4.106 0 001.27 5.477A4.072 4.072 0 012.8 9.713v.052a4.105 4.105 0 003.292 4.022 4.095 4.095 0 01-1.853.07 4.108 4.108 0 003.834 2.85A8.233 8.233 0 012 18.407a11.616 11.616 0 006.29 1.84"
            />
          </svg>
        </div>
      </a>
      <a
        href="https://github.com/codemanufacture"
        tw="text-gray-400 hover:text-gray-500"
      >
        <span
          tw="sr-only"
        >
          GitHub
        </span>
        <div
          tw="h-6 w-6"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </a>
    </div>
    <p
      tw="mt-10 text-center text-xs leading-5 text-gray-500"
    >
      Â© 
      2023
       Codemanufacture SRLS. All rights reserved. Partita iva: 12607110967
    </p>
  </div>
</footer>
`;
